bears/general/KeywordBear.py

```diff
@@ -2,18 +2,71 @@
 
 from coalib.bearlib import deprecate_settings
 from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
 from coalib.results.Result import RESULT_SEVERITY, Result
 
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.results.SourceRange import SourceRange
+
+
+def _get_comments(dependency_results):
+    if not dependency_results:
+        return []
+    annotation_bear_results = dependency_results.get('AnnotationBear')
+    if (not annotation_bear_results or
+            not isinstance(annotation_bear_results, list)):
+        return []
+    comments = []
+    for result in annotation_bear_results:
+        if not isinstance(result.contents, dict):
+            continue
+        comments += list(result.contents.get('comments', []))
+    return set(comments)
+
+
+def get_diff_comment_todo(dependency_results, file, filename,
+                          line, line_number, pos):
+    comments = _get_comments(dependency_results)
+    diffs = None
+    for source_range_comment in comments:
+        todo_source_range = SourceRange.from_values(
+            filename, line_number + 1, pos + 1)
+        if todo_source_range not in source_range_comment:
+            continue
+
+        diff = Diff(file)
+        # remove just part of the line
+        comment_start = source_range_comment.start.column
+        line_before_todo_comment = line[:comment_start - 1].strip()
+        if (line_before_todo_comment and
+                line_number + 1 == source_range_comment.start.line):
+            diff.change_line(
+                line_number + 1,
+                line,
+                line.replace(line[comment_start - 1:],
+                             '').rstrip() + '\n'
+            )
+        else:
+            diff.delete_line(line_number + 1)
+
+        diffs = {filename: diff}
+    return diffs
+
 
 class KeywordBear(LocalBear):
     LANGUAGES = {"All"}
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     LICENSE = 'AGPL-3.0'
     CAN_DETECT = {'Documentation'}
+    BEAR_DEPS = {AnnotationBear}
 
     @deprecate_settings(keywords='ci_keywords')
-    def run(self, filename, file, keywords: list):
+    def run(self,
+            filename,
+            file,
+            keywords: list,
+            dependency_results: dict = None):
```

Comment:
we usually don't want spaces around the `=`, PEP8Bear doesn't catch that sadly...